src/opm/output/eclipse/CreateInteHead.cpp

```diff
@@ -0,0 +1,234 @@
+
+        const auto& phasePred = rspec.phases();
+
+        phases.oil   = phasePred.active(Opm::Phase::OIL);
+        phases.water = phasePred.active(Opm::Phase::WATER);
+        phases.gas   = phasePred.active(Opm::Phase::GAS);
+
+        return phases;
+    }
+
+    Opm::RestartIO::InteHEAD::TuningPar
+    getTuningPars(const ::Opm::Tuning& tuning,
+                  const std::size_t    step)
+    {
+        const auto& newtmx = tuning.getNEWTMX(step);
+        const auto& newtmn = tuning.getNEWTMN(step);
+        const auto& litmax = tuning.getLITMAX(step);
+        const auto& litmin = tuning.getLITMIN(step);
+        const auto& mxwsit = tuning.getMXWSIT(step);
+        const auto& mxwpit = tuning.getMXWPIT(step);
+
+        return {
+            newtmx,
+            newtmn,
+            litmax,
+            litmin,
+            mxwsit,
+            mxwpit,
+        };
+    }
+
+    Opm::RestartIO::InteHEAD::WellSegDims
+    getWellSegDims(const ::Opm::Runspec&  rspec,
+                   const ::Opm::Schedule& sched,
+                   const std::size_t      step)
+    {
+        const auto& wsd = rspec.wellSegmentDimensions();
+
+        const std::size_t step_minus_one = step == 0 ? 0 : step - 1;
+        const auto& sched_wells = sched.getWells(step_minus_one);
+
+        const auto nsegwl =
+            std::count_if(std::begin(sched_wells), std::end(sched_wells),
+                [step_minus_one](const Opm::Well* wellPtr)
+            {
+                return wellPtr->isMultiSegment(step_minus_one);
+            });
+
+        const auto nswlmx = wsd.maxSegmentedWells();
+        const auto nsegmx = wsd.maxSegmentsPerWell();
+        const auto nlbrmx = wsd.maxLateralBranchesPerWell();
+        const auto nisegz = 22;
+        const auto nrsegz = 140;
+        const auto nilbrz = 10;
+
+        return {
+            static_cast<int>(nsegwl),
+            nswlmx,
+            nsegmx,
+            nlbrmx,
+            nisegz,
+            nrsegz,
+            nilbrz
+        };
+    }
+
+    Opm::RestartIO::InteHEAD::RegDims
+    getRegDims(const ::Opm::TableManager& tdims,
+               const ::Opm::Regdims&      rdims)
+    {
+        const auto ntfip  = tdims.numFIPRegions();
+        const auto nmfipr = rdims.getNMFIPR();
+        const auto nrfreg = rdims.getNRFREG();
+        const auto ntfreg = rdims.getNTFREG();
+        const auto nplmix = rdims.getNPLMIX();
+
+        return {
+            static_cast<int>(ntfip),
+            static_cast<int>(nmfipr),
+            static_cast<int>(nrfreg),
+            static_cast<int>(ntfreg),
+            static_cast<int>(nplmix),
+        };
+    }
+} // Anonymous
+
+// #####################################################################
+// Public Interface (createInteHead()) Below Separator
+// ---------------------------------------------------------------------
+
+std::vector<int>
+Opm::RestartIO::Helpers::
+createInteHead(const EclipseState& es,
```

Comment:
I would prefer the form:
```C++
std::vector<int> Opm::RestartIO::Helpers::createIntehaed( ) {
}
```

Reply:
On the other hand, that leads to long lines.  I tend to prefer code that's readable in 80 character terminals and otherwise sticks close to the left margin.